src/components/LoadingButton.jsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default function LoadingButton(props) {
+  // extract props
+  const {
+    label,
+    disabled,
+    disabledStyle,
+    className,
+    callback,
+    ...remainingProps
+  } = props;
+
+  // track whether button is loading or not
+  const [isLoading, setButtonLoading] = React.useState(false);
+  return (
+    <button
+      type="button"
+      className={isLoading ? disabledStyle : className}
+      disabled={isLoading ? true : disabled}
+      onClick={() => {
+        setButtonLoading(true);
+        callback().then(() => setButtonLoading(false));
```

Comment:
Maybe use `await` to keep the style consistent with the rest of the code when dealing with async funcs

Reply:
done